src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,142 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+
+/**
+ * Parses and understands user commands.
+ * Break down user input string and creates appropriate user commands.
+ * Checks for wrong user inputs and throw Duke Exceptions if necessary.
+ */
+public class Parser {
+
+    /**
+     * Static method to parse user input.
+     * @param userCommand User string from the command line.
+     * @return Appropriate command to reflect user input.
+     * @throws DukeException
+     */
+    public static Command parse(String userCommand) throws DukeException {
+
+        // Formats the way users input their dates for a task.
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd.MM.yy");
+
+        // Formats the way users input their time for a task.
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
+        String[] userWord = userCommand.split(" ", 2);
+
+        // Exits the program.
+        if (userCommand.equals("bye")) {
+            return new ExitCommand();
+        }
+        // Switch block checks for the appropriate commands to create based on user input.
+        switch (userWord[0]) {
+        case "list":
+            return new ListCommand();
+            //Fallthrough
+
+        case "find" :
+            if (userWord.length == 1 || userWord[1].equals("")) {
+                throw new DukeException("   ☹ OOPS!!! The description of the command find cannot be empty.");
+            }
+            return new FindCommand(userWord[1]);
+            //Fallthrough
+
+        case "todo":
+            if (userWord.length == 1 || userWord[1].equals("")) {
+                throw new DukeException("   ☹ OOPS!!! The description of the command todo cannot be empty.");
+            }
+            return new AddCommand("todo", userWord[1]);
+            //Fallthrough
+
+        case "deadline":
+            if (userWord.length == 1 || userWord[1].equals("")) {
+                throw new DukeException("   ☹ OOPS!!! The description of the command deadline cannot be empty.");
+            }
+            String[] content = userWord[1].split("/by", 2);
+            if (content.length == 1 || content[0].equals("")) {
+                throw new DukeException("   ☹ OOPS!!! We can't seem to find your event description.");

```

Comment:
I think it is better to standardise the format of DukeException? I've noticed the DukeExceptions thrown in Command classes did not have the sad faces. Perhaps you can abstract '   ☹ OOPS!!!' out into DukeException itself so you don't have to type it repetitively throughout your code?